slack-java-client/src/main/java/com/hubspot/slack/client/ratelimiting/ByMethodRateLimiter.java

```diff
@@ -0,0 +1,20 @@
+package com.hubspot.slack.client.ratelimiting;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import com.google.common.util.concurrent.RateLimiter;
+import com.hubspot.slack.client.methods.SlackMethod;
+
+public class ByMethodRateLimiter implements SlackRateLimiter {
```

Comment:
This should almost certainly be annotated as `@Singleton` to avoid having many instances around (since it's backed by a shared map).